### ISSUE TITLE
ProtectHostname implies NoNewPrivileges

### ISSUE THREAD

**Issue by dechamps** [systemd/systemd#18914]
**systemd version the issue has been seen with**
247.3-1

**Used distribution**
Debian Unstable

**Linux kernel version used** (`uname -a`)
`5.10.0-4-amd64 #1 SMP Debian 5.10.19-1 (2021-03-02) x86_64 GNU/Linux`

**Expected behaviour you didn't see**
Documentation says nothing about [`ProtectHostname=`](https://www.freedesktop.org/software/systemd/man/systemd.exec.html#ProtectHostname=) interacting with [`NoNewPrivileges=`](https://www.freedesktop.org/software/systemd/man/systemd.exec.html#NoNewPrivileges=), so I expected them to be independent.

**Unexpected behaviour you saw**
`ProtectHostname=yes` appears to imply `NoNewPrivileges=yes`.

**Steps to reproduce the problem**
```
# cp /usr/bin/id /tmp/
# chmod +s /tmp/id
# cat >/etc/systemd/system/test-suid.service <<EOF
[Service]
ExecStart=/tmp/id --user
User=nobody
Group=nogroup

ProtectHostname=yes
EOF
# systemctl start test-suid.service
# journalctl -u test-suid.service
```

The output will be:

```
systemd[1]: Started test-suid.service.
id[…]: 65534
systemd[1]: test-suid.service: Succeeded.
```

The effective UID is `65534` (nobody), as if `NoNewPrivileges` were used. I expected `0` (root), and this is what is output if `ProtectHostname=yes` is removed.

**Comment by yuwata** [comment]
Right. The man page needs to be updated. https://github.com/systemd/systemd/blob/63dc82d378e1cfda8f560e8da9aa2df549c2b026/src/core/execute.c#L1423-L1448

**Comment by yuwata** [comment]
Fix is waiting in #18918.
